src/jit/lclvars.cpp

```diff
@@ -3292,6 +3292,13 @@ void Compiler::lvaSortByRefCount()
         {
             varDsc->lvTracked = 0;
         }
+#if defined(JIT32_GCENCODER) && defined(WIN64EXCEPTIONS)
+        else if (lvaIsOriginalThisArg(lclNum))
+        {
+            // For x86/Linux, we cannot have this pointer in tracked variables, so we set this pointer untracked
+            varDsc->lvTracked = 0;
```

Comment:
Isn't this just part of the contract between JIT and VM? That is, JIT only reports `this` in the GC information in exactly the cases that VM requires? If `lvaKeepAliveAndReportThis()` captures that condition, then it seems like that would be sufficient.

Can you explain why you think the JIT needs to ask a VM a question, or why we need a GCInfo bit? (E.g., what would the bit say?)

---
In reply to: [117710335](https://github.com/dotnet/coreclr/pull/11281#discussion_r117710335) [](ancestors = 117710335)